src/render/draw_raster.js

```diff
@@ -100,3 +105,23 @@ function drawRaster(painter: Painter, sourceCache: SourceCache, layer: RasterSty
     painter.resetStencilClippingMasks();
 }
 
+function configureRasterColor (layer: RasterStyleLayer, context: Context, gl: WebGLRenderingContext) {
+    const defines = [];
+    let mix;
+    let range;
+    let unit = -1;
+
+    if (layer.paint.get('raster-color')) {
+        defines.push('RASTER_COLOR');
+        mix = layer.paint.get('raster-color-mix');
+        range = layer.paint.get('raster-color-range');
+
+        // Allocate a texture if not allocated
+        context.activeTexture.set(gl.TEXTURE2);
+        unit = 2;
```

Comment:
If we only pass 2 for `u_color_ramp` do we really need an uniform for it or could we just hardcode it in the shader? Or is it needed for some future use-cases?

Reply:
Upon further investigation, you can't hard-code the sampler index inside the shader. It has to be a uniform. I've at least moved this to a js `const` declaration. Note that per previous discussion, WebGL is smart enough not to set uniforms if the uniform is inside an inactive preprocessor declaration, so that it at least shouldn't have meaningful extra cost.